### ISSUE TITLE
Move discourse.webmaker.org to new domain

### ISSUE THREAD

**Issue by ldecoursy** [MozillaFoundation/mofo-devops#253]
We need to update the URL for the Discourse forum linked to teach.mozilla.org so it no longer lives on the webmaker domain–currently it's at http://discourse.webmaker.org

@simonwex offered to help explore our options and delegate next steps

cc @hannahkane @kgorr

**Comment by simonwex** [comment]
@ldecoursy it's not going to be possible to put this on a mozilla.org domain. We can register a new top-level domain for this, do you have any ideas? 

Let's scratch it out in this document: https://docs.google.com/a/mozilla.com/document/d/1NA5rYu2eVsTnCIv8VWAqyHhcn5bLF2SNrH4UTbi2AiU/edit?usp=sharing

(don't leave domain suggestions here as someone else could buy the domain on us)

**Comment by ldecoursy** [comment]
Left a few suggestions - would also love input from @chrislarry33 and @hannahkane

**Comment by ldecoursy** [comment]
@simonwex Just checking back on this one.

**Comment by hannahkane** [comment]
Sorry I missed this mention earlier. Just commented on the doc.

**Comment by cadecairos** [comment]
@simonwex update?

**Comment by cadecairos** [comment]
@hannahkane did we settle on ` forum.teach.mozilla.org` then?

**Comment by hannahkane** [comment]
Thanks for the ping, @cadecairos.

teach.mozilla.org is scheduled to move to learning.mozilla.org at some point. See https://github.com/MozillaFoundation/plan/issues/764.

@ldecoursy @simonwex - should we wait for that move to happen first before doing this?

**Comment by cadecairos** [comment]
Note to self: `csa-ttw-production-632305050` is the load balancer to update once we know where this is moving.

**Comment by ldecoursy** [comment]
@hannahkane Do we have an ETA for when that move might happen?

**Comment by hannahkane** [comment]
cc @xmatthewx who's on that /plan ticket. I'd say we should do this sooner rather than later, but I may not be thinking about all of the repercussions, both technically and perhaps in terms of user communication. Will start a separate thread about this on the [plan ticket](https://github.com/MozillaFoundation/plan/issues/764)

**Comment by flamingspaz** [comment]
@cadecairos the container will also need to be rebuilt or emails will send out with the old domain, lmk when the new domain is pointing at that lb and I can make the changes.

**Comment by cadecairos** [comment]
I think we should call this one blocked on:

- [ ] #310

**Comment by hannahkane** [comment]
@cadecairos - now that this is unblocked, should we put this in the next heartbeat? I know you triage on Mondays, so no need to reply until after that.

**Comment by cadecairos** [comment]
@ldecoursy I just created an account on discourse.webmaker.org, are you able to promote my account to admin?

username is `cade`
email is `cade@mozillafoundation.org`

**Comment by ldecoursy** [comment]
Admin has been granted 🙇

**Comment by cadecairos** [comment]
Judging from what I've just seen in the admin interface, and read in the docs, There's isn't anything to update using that tool.

Once we switch over to discourse.learning.mozilla.org, I'll have to update the Webmaker OAuth callback for logging in, and I'll likely have to update a couple environment variables on the discourse server itself - although I do not know how they (environment variables for discourse) currently managed.

The investigation continues!

**Comment by xmatthewx** [comment]
As discussed in Chat, `forum` works for me. I'm slightly concerned about the daisy-chain of subdomains – `forum.learning.mozilla.org` – the public usually sees those only in weird corners of government services. But, the change will be a step in the right direction. As @cadecairos suggested, we could redirect from `learning.mozilla.org/forum` for anywhere we display the URL.

**Comment by flamingspaz** [comment]
Okay, I've rebuilt the Discourse container with the new domain.

@cadecairos all that's left is to change the id.wmo callback and change the DNS. If you cc me in the DNS bug I can change the SSL cert as soon as the DNS is changed so people don't get errors.

**Comment by cadecairos** [comment]
Need to figuyre out if MoCo want's to do a security review before we host this on mozilla.org

**Comment by cadecairos** [comment]
filed a bug to add a CNAME for forum.learning.mozilla.org - https://bugzilla.mozilla.org/show_bug.cgi?id=1279527

@flamingspaz your bugmail account is cc'd so you'll be able to follow the progress there.

**Comment by cadecairos** [comment]
@flamingspaz how are we looking here?

**Comment by cadecairos** [comment]
Done.

https://discourse.webmaker.org is redirecting to https://forum.learning.mozilla.org

Fixed the redirect url on id, so webmaker login still works.

**Comment by KGorr** [comment]
Hey @cadecairos - did the digest email template get updated too? I looked at the settings on my to see but can't tell what I am looking at.

**Comment by cadecairos** [comment]
@flamingspaz is going to rebuild (or has already rebuilt) the app so that the email settings reflect the URL change. However, I've set up the old url to redirect to the new site (preserving paths and query params)

**Comment by KGorr** [comment]
Thanks!

**Comment by flamingspaz** [comment]
All done, everything should be :ok_hand:

We also took this opportunity to remove persona as it's always been buggy and will be shut down soon.

**Comment by xmatthewx** [comment]
Do we need to send a message to users on how to login without persona? @simonwex @ldecoursy

**Comment by cadecairos** [comment]
We could. the only option now is to use Webmaker OAuth

**Comment by KGorr** [comment]
My vote is to send communication on the log in changes to avoid confusion. Please let me know if you need help! @xmatthewx @cadecairos

**Comment by xmatthewx** [comment]
@KGorr - do you mind filing a new issue in this repo? Someone probably has a prior message we could repurpose.

**Comment by KGorr** [comment]
Done! Hope it's clear: https://github.com/MozillaFoundation/mofo-devops/issues/372 @xmatthewx
